apps/server/src/modules/board/controller/api-test/drawing-item-check-permission.api.spec.ts

```diff
@@ -0,0 +1,118 @@
+		await app.init();
+		em = module.get(EntityManager);
+		testApiClient = new TestApiClient(app, baseRouteName);
+	});
+
+	afterAll(async () => {
+		await app.close();
+	});
+
+	describe('when user is a valid teacher who is part of course', () => {
+		const setup = async () => {
+			await cleanupCollections(em);
+
+			const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher();
+			const course = courseFactory.build({ teachers: [teacherUser] });
+			await em.persistAndFlush([teacherAccount, teacherUser, course]);
+
+			const columnBoardNode = columnBoardNodeFactory.buildWithId({
+				context: { id: course.id, type: BoardExternalReferenceType.Course },
+			});
+
+			const columnNode = columnNodeFactory.buildWithId({ parent: columnBoardNode });
+
+			const cardNode = cardNodeFactory.buildWithId({ parent: columnNode });
+
+			const drawingItemNode = drawingElementNodeFactory.buildWithId({ parent: cardNode });
+
+			await em.persistAndFlush([columnBoardNode, columnNode, cardNode, drawingItemNode]);
+			em.clear();
+
+			const loggedInClient = await testApiClient.login(teacherAccount);
+
+			return { loggedInClient, teacherUser, columnBoardNode, columnNode, cardNode, drawingItemNode };
+		};
+
+		it('should return status 201', async () => {
+			const { loggedInClient, drawingItemNode } = await setup();
+
+			const response = await loggedInClient.get(`${drawingItemNode.id}/permission`);
+
+			expect(response.status).toEqual(201);
+		});
+	});
+
+	describe('when only teacher is part of course', () => {
+		const setup = async () => {
+			await cleanupCollections(em);
+
+			const { teacherAccount, teacherUser } = UserAndAccountTestFactory.buildTeacher();
+			const { studentAccount, studentUser } = UserAndAccountTestFactory.buildStudent();
+
+			const course = courseFactory.build({ students: [teacherUser] });
+			await em.persistAndFlush([teacherAccount, teacherUser, course, studentAccount, studentUser]);
+
+			const columnBoardNode = columnBoardNodeFactory.buildWithId({
+				context: { id: course.id, type: BoardExternalReferenceType.Course },
+			});
+
+			const columnNode = columnNodeFactory.buildWithId({ parent: columnBoardNode });
+
+			const cardNode = cardNodeFactory.buildWithId({ parent: columnNode });
+
+			const drawingItemNode = drawingElementNodeFactory.buildWithId({ parent: cardNode });
+
+			await em.persistAndFlush([columnBoardNode, columnNode, cardNode, drawingItemNode]);
+			em.clear();
+
+			const loggedInClient = await testApiClient.login(studentAccount);
+
+			return { loggedInClient, studentUser, columnBoardNode, columnNode, cardNode, drawingItemNode };
+		};
+
+		it('should return status 401 for student not assigned to course', async () => {
+			const { loggedInClient, drawingItemNode } = await setup();
+
+			const response = await loggedInClient.get(`${drawingItemNode.id}/permission`);
+
+			expect(response.status).toEqual(403);
+		});
+
+		it('should return status 404 for wrong id', async () => {
```

Comment:
This test does not depend upon the user. Putting it inside the scenario "when only teacher is part of course" is misleading.